### PR TITLE
Adding CodeLlama-7B inference and compilation example notebook

### PR DESCRIPTION
**Pull request by jimburtoft**
# What does this PR do?
Adds a CodeLlama-7B inference example.

## Before submitting
- [ x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by jimburtoft** [comment]
We're working to get some more model specific examples out using optimum-neuron.  

I did NOT include a link in docs/source/tutorials/overview.mdx nor do I have an mdx version of the notebook.  You are welcome to convert it or I can add it if you want to send instructions.

File: notebooks/text-generation/CodeLlama-7B-Compilation.ipynb [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    "1. Clone the [AWS Neuron Samples](https://github.com/aws-neuron/aws-neuron-samples) repo to your instance using\n",
+    "```\n",
+    "git clone https://github.com/aws-neuron/aws-neuron-samples.git\n",
+    "```\n",
+    "2. Navigate to the `transformers-neuronx` inference samples folder\n",
+    "```\n",
+    "cd aws-neuron-samples/torch-neuronx/transformers-neuronx/inference\n",
+    "```\n",
+    "3. Follow the instructions in [Jupyter Notebook QuickStart](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/general/setup/notebook/setup-jupyter-notebook-steps-troubleshooting.html) to run Jupyter Notebook on your instance.\n",
+    "4. Locate this tutorial in your Jupyter Notebook session (`CodeLlama-7B.ipynb`) and launch it. Follow the rest of the instructions in this tutorial. "
```
Comment (dacorvo) [inline notebooks/text-generation/CodeLlama-7B-Compilation.ipynb:70]: I believe this should be fetch from optimum-neuron instead, isn't it ?

Comment (jimburtoft) [reply notebooks/text-generation/CodeLlama-7B-Compilation.ipynb:70]: Yes, I'll fix it.

Comment (jimburtoft) [reply notebooks/text-generation/CodeLlama-7B-Compilation.ipynb]: @dacorvo I fixed a number of path issues in that section.  Thanks for the fresh eyes.

**Review by dacorvo — APPROVED** [review]
LGTM, thanks for the pull-request !

**Pull request merged**